build/maven/xmlcodegen/src/main/java/org/geotools/maven/xmlcodegen/AbstractGeneratorMojo.java

```diff
@@ -299,100 +299,87 @@ protected XSDSchema schema() {
 		}
 		
 		//add a schema locator which uses the xsd objects to get at the schemas
-		XSDSchemaLocator locator = new XSDSchemaLocator() {
-
-            public XSDSchema locateSchema(XSDSchema schema, String namespaceURI,
-                String rawSchemaLocationURI, String resolvedSchemaLocationURI) {
-
-                for (XSD xsd : xsds) {
-                    if (xsd == null) {
-                        continue;
-                    }
-                    if (xsd.getNamespaceURI().equals(namespaceURI)) {
-                        try {
-                            return xsd.getSchema();
-                        } catch (IOException e) {
-                            getLog().warn("XSDSchemaLocator: Error occurred locating schema: " + namespaceURI, e);
-                        }
-                    }
-                }
-             
-                getLog().warn( "XSDSchemaLocator: Could not locate schema for: " + namespaceURI );
-                return null;
-            }
-		    
-		};
+		XSDSchemaLocator locator = (schema, namespaceURI, rawSchemaLocationURI,
```

Comment:
I personally find these two pretty ugly too, but they were bad in the original as well.. such a large anonymous class shold probably be turned into an inner class.

Reply:
Yep a net zero improvement on this one.